tempesta_fw/t/functional/helpers/control.py

```diff
@@ -15,6 +16,46 @@
 # Clients
 #-------------------------------------------------------------------------------
 
+STATE_BEGIN_START = "begin_start"
+STATE_STARTED = "started"
+STATE_STOPPED = "stopped"
+STATE_ERROR = "error"
+
+class Stateful(object):
+    state = STATE_STOPPED
+    stop_procedures = []
+
+    def run_start(self):
+        """ Shoul be overrided """
+        pass
+
+    def start(self):
+        """ Try to start object """
+        if self.state != STATE_STOPPED:
+            return
+        self.state = STATE_BEGIN_START
```

Comment:
It's possible to initialize `self.state` with `ERROR` state. In this case special processing for the `STATE_BEGIN_START` won't be needed and one less state to worry about. What do think?

Reply:
I think it is a bad idea.

Reply:
It can lead to unexpected behaviour

Reply:
Testing framework it is a bad place for some 'hacks' and optimizations. It must be straightforward

Reply:
When we begin start procedure, we are not in error state

Reply:
So we must not set state to error